docs/content/guide/i18n.ngdoc

```diff
@@ -142,84 +144,179 @@ displaying the date with a timezone specified by the developer.
-later.)
+### Plural syntax with optional `offset`
+
+The plural syntax supports an optional `offset` syntax that is used in matching.  It's simpler to
+explain this with an example.
+
+```text
+{{recipients.length, plural, offset:1
+    =0    {You gave no gifts}
+    =1    {You gave {{recipients[0].name}} a gift}
+    one   {You gave {{recipients[0].name}} and one other person a gift}
+    other {You gave {{recipients[0].name}} and # other people a gift}
+}}
+```
+
+When an `offset` is specified, the matching works as follows.  First, the exact value of the Angular
+expression is matched against the exact matches (i.e. `=N` selectors) to find a match.  If there is
+one, that message is used.  If there was no match, then the offset value is subtracted from the
+value of the expression and locale specific pluralization rules are applied to this new value to
+obtain its plural category (such as “one”, “few”, “many”, etc.) and a match is attempted against the
+keyword selectors and the matching message is used.  If there was no match, then the “other”
+category (required) is used.  The value of the `#` character inside a message is the value of
+original expression reduced by the offset value that was specified.
+
+### Escaping / Quoting
+
+You might need to quote the curly braces or the `#` character inside the message texts so that they
+are treated literally with no special meaning.  To support such cases, you may escape any character
+by preceding it with a `\` (backslash) character.  The backslash character removes any special
+meaning to the character that immediately follows it.  Therefore, you can escape or quote the
+backslash itself by preceding it with another backslash character.
 
 
-### Simple select (for gender) example
+### Gender (aka select) Syntax
 
+The gender support is provided by the more generic "select" syntax that is more akin to a switch
+statement.  It is general enough to support use for gender based messages.
+
+The syntax for gender based message selection looks like this following:
+
+```text
+{{EXPRESSION, select,
+    male {MESSAGE_WHEN_EXPRESSION_IS_MALE}
+    female {MESSAGE_WHEN_EXPRESSION_IS_FEMALE}
+    ...
+    other {MESSAGE_WHEN_THERE_IS_NO_GENDER_MATCH}
+}}
 ```
+
+Please note that whitespace (including newline) is generally insignificant except as part of the
+actual message text that occurs in curly braces.  Whitespace is generally used to aid readability.
+
+Here, `EXPRESSION` is an Angular expression that evaluates to the gender of the person that
+is used to select the message that should be displayed.
+
+The Angular expression is followed by `, select,` where the spaces are optional.
+
+This is followed by a list of selection keyword and corresponding message pairs.  The "other"
+keyword and corresponding message are **required** but you may have as few or as many of the other
+gender values as you need (i.e. it isn't restricted to male/female.)  Note however, that the
+matching is **case-sensitive**.
```

Comment:
Sadly, that's not an option. :)  That just seems to be how MessageFormat has been defined.  We could deviate from the spec and use something else instead of "other" for the default case, but I feel like that is a bit too much for a rare case.  (Our other deviations are more understandable.)